app/controllers/form_items_controller.rb

```diff
@@ -0,0 +1,19 @@
+class FormItemsController < ApplicationController
+  # authorization via cancan
+  load_and_authorize_resource
+
+  def update
+    if params[:parent_id].blank? or !@form_item.check_ancestry_integrity(params[:parent_id])
+      return render json: { errors: ['ancestry_check_failed'] }, status: 422
+    end
+
+    @form_item.ancestry = params[:parent_id]
```

Comment:
This won't quite do it. For one, I think it's bad to set ancestry directly because not sure if stuff like the depth cache would get set properly then. Also if the parent is not root, this won't work. Instead, try `@form_item.parent = FormItem.find(params[:parent_id])`?